src/preferences/broadcastprofile.cpp

```diff
@@ -0,0 +1,466 @@
+    XmlParse::addElement(doc, docRoot, kServertype, m_serverType);
+    XmlParse::addElement(doc, docRoot, kLogin, m_login);
+    XmlParse::addElement(doc, docRoot, kPassword, m_password);
+
+    XmlParse::addElement(doc, docRoot, kEnableReconnect,
+                         QString::number((int)m_enableReconnect));
+    XmlParse::addElement(doc, docRoot, kReconnectPeriod,
+                         QString::number(m_reconnectPeriod));
+
+    XmlParse::addElement(doc, docRoot, kLimitReconnects,
+                         QString::number((int)m_limitReconnects));
+    XmlParse::addElement(doc, docRoot, kMaximumRetries,
+                         QString::number(m_maximumRetries));
+
+    XmlParse::addElement(doc, docRoot, kNoDelayFirstReconnect,
+                         QString::number((int)m_noDelayFirstReconnect));
+    XmlParse::addElement(doc, docRoot, kReconnectFirstDelay,
+                         QString::number(m_reconnectFirstDelay));
+
+    XmlParse::addElement(doc, docRoot, kMountPoint, m_mountpoint);
+    XmlParse::addElement(doc, docRoot, kStreamName, m_streamName);
+    XmlParse::addElement(doc, docRoot, kStreamDesc, m_streamDesc);
+    XmlParse::addElement(doc, docRoot, kStreamGenre, m_streamGenre);
+    XmlParse::addElement(doc, docRoot, kStreamPublic,
+                         QString::number((int)m_streamPublic));
+    XmlParse::addElement(doc, docRoot, kStreamWebsite, m_streamWebsite);
+
+    XmlParse::addElement(doc, docRoot, kFormat, m_format);
+    XmlParse::addElement(doc, docRoot, kBitrate,
+                         QString::number(m_bitrate));
+    XmlParse::addElement(doc, docRoot, kChannels,
+                         QString::number(m_channels));
+
+    XmlParse::addElement(doc, docRoot, kEnableMetadata,
+                         QString::number((int)m_enableMetadata));
+    XmlParse::addElement(doc, docRoot, kMetadataCharset, m_metadataCharset);
+    XmlParse::addElement(doc, docRoot, kCustomArtist, m_customArtist);
+    XmlParse::addElement(doc, docRoot, kCustomTitle, m_customTitle);
+    XmlParse::addElement(doc, docRoot, kMetadataFormat, m_metadataFormat);
+    XmlParse::addElement(doc, docRoot, kOggDynamicUpdate,
+                         QString::number((int)m_oggDynamicUpdate));
+
+    doc.appendChild(docRoot);
+
+    QFile xmlFile(filename);
+    xmlFile.open(QIODevice::WriteOnly | QIODevice::Text);
+
+    QTextStream fileStream(&xmlFile);
+    doc.save(fileStream, 4);
+    xmlFile.close();
+}
+
+void BroadcastProfile::setProfileName(const QString &profileName) {
+    m_profileName = QString(profileName);
+
+    // Replace occurences of forbidden characters with a space
+    // to avoid problems with the underlying filesystem.
+    // Char list comes from MSDN article "Naming Files, Paths, and Namespaces".
+    m_profileName.replace(kForbiddenChars, " ");
+}
+
+QString BroadcastProfile::getProfileName() const {
+    return m_profileName;
+}
+
+bool BroadcastProfile::getEnabled() const {
+    return m_enabled;
```

Comment:
It was already there in the previous BroadcastSettings, so I kept it because it might come in handy for multi-broadcasting (this would mean: "this connection is enabled").